service/src/main/java/tennisclub/facade/EventFacadeImpl.java

```diff
@@ -38,15 +38,16 @@ public EventFacadeImpl(EventService eventService, CourtService courtService, Map
     }
 
     @Override
-    public void reschedule(EventRescheduleDTO eventRescheduleDTO) {
-        Event event = eventService.findById(eventRescheduleDTO.getId());
+    public EventWithCourtDTO reschedule(Long id, EventRescheduleDTO eventRescheduleDTO) {
+        Event event = eventService.findById(id);
         Court court = event.getCourt();
 
         if (!courtService.isFree(court, eventRescheduleDTO.getStart(), eventRescheduleDTO.getEnd())) {
             throw new FacadeLayerException("Can't reschedule the event. The court is not free at the new time.");
         }
 
-        eventService.reschedule(event, eventRescheduleDTO.getStart(), eventRescheduleDTO.getEnd());
+        Event updated = eventService.reschedule(event, eventRescheduleDTO.getStart(), eventRescheduleDTO.getEnd());
+        return mapper.map(updated, EventWithCourtDTO.class);
```

Comment:
This should be using the private `map` method of the facade since this way we lose the information contained in the subclasses of `Event`.

Reply:
Thank you for objevení da chyba❤️